internal/ad/ad.go

```diff
@@ -389,6 +409,57 @@ func (ad *AD) ensureKrb5CCName(srcKrb5CCName, dstKrb5CCName string) (err error)
 	return nil
 }
 
+// ensureKrb5CCCopy checks if the cached krb5cc ticket is up to date and updates
+// it if not.
+func ensureKrb5CCCopy(srcKrb5CCName, krb5CCCopyName string) (err error) {
```

Comment:
Same thing on the error decoration here...

Maybe it's better to stop decorating on the "helper" functions and decorate only the error on `ensureKrb5CCName` to avoid too much decorate chains?